src/protocol.c

```diff
@@ -102,9 +102,31 @@ static bool spawn_process(struct pss_tty *pss, uint16_t columns, uint16_t rows)
   for (i = 0; i < server->argc; i++) {
     argv[n++] = server->argv[i];
   }
-  for (i = 0; i < pss->argc; i++) {
-    argv[n++] = pss->args[i];
+  if (server->url_arg) {
+    for (i = 0; i < pss->argc; i++) {
+      argv[n++] = pss->args[i];
+    }
   }
+  else if (server->arg_file != NULL) {
+    int fd = -1;
+    char *filePath = strcat(server->arg_file, "XXXXXX");
```

Comment:
I think we should just choose a prefix here and not require users to choose a prefix

Reply:
also this is not a safe usage of strcat, since you may write past the allocated memory